pkg/cloud/services/ec2/securitygroups.go

```diff
@@ -374,11 +375,11 @@ func (s *Service) getSecurityGroupIngressRules(role infrav1.SecurityGroupRole) (
 		return infrav1.IngressRules{
 			s.defaultSSHIngressRule(s.scope.SecurityGroups()[infrav1.SecurityGroupBastion].ID),
 			{
-				Description: "Kubernetes API",
-				Protocol:    infrav1.SecurityGroupProtocolTCP,
-				FromPort:    6443,
-				ToPort:      6443,
-				CidrBlocks:  []string{anyIPv4CidrBlock},
+				Description:            "Kubernetes API",
+				Protocol:               infrav1.SecurityGroupProtocolTCP,
+				FromPort:               6443,
+				ToPort:                 6443,
+				SourceSecurityGroupIDs: []string{s.scope.SecurityGroups()[infrav1.SecurityGroupAPIServerLB].ID},
```

Comment:
This might effect how the Talos folks are doing bootstrapping. tagging in @rsmitty

Reply:
Line 382 is a mistake. It should continue to be open to `anyIPv4CidrBlock`.  Not sure what I was thinking at the time. Will change it back.

Reply:
reverted the mistaken change on line 382.